pallets/lbp/src/tests.rs

```diff
@@ -0,0 +1,1680 @@
+			final_weights: ((ACA, 90), (DOT, 10)),
+			last_weight_update: 0u64,
+			last_weights: ((ACA, 20), (DOT, 80)),
+			curve: WeightCurveType::Linear,
+			pausable: true,
+			paused: false,
+		};
+		assert_ok!(LBPPallet::validate_pool_data(&pool_data));
+
+		// null interval
+		let pool_data = Pool {
+			start: 0u64,
+			end: 0u64,
+			initial_weights: ((ACA, 20), (DOT, 80)),
+			final_weights: ((ACA, 90), (DOT, 10)),
+			last_weight_update: 0u64,
+			last_weights: ((ACA, 20), (DOT, 80)),
+			curve: WeightCurveType::Linear,
+			pausable: true,
+			paused: false,
+		};
+		assert_ok!(LBPPallet::validate_pool_data(&pool_data));
+
+		let pool_data = Pool {
+			start: 10u64,
+			end: 2u64,
+			initial_weights: ((ACA, 20), (DOT, 80)),
+			final_weights: ((ACA, 90), (DOT, 10)),
+			last_weight_update: 0u64,
+			last_weights: ((ACA, 20), (DOT, 80)),
+			curve: WeightCurveType::Linear,
+			pausable: true,
+			paused: false,
+		};
+		assert_noop!(
+			LBPPallet::validate_pool_data(&pool_data),
+			Error::<Test>::InvalidBlockNumber
+		);
+
+		let pool_data = Pool {
+			start: 10u64,
+			end: 11u64 + u32::MAX as u64,
+			initial_weights: ((ACA, 20), (DOT, 80)),
+			final_weights: ((ACA, 90), (DOT, 10)),
+			last_weight_update: 0u64,
+			last_weights: ((ACA, 20), (DOT, 80)),
+			curve: WeightCurveType::Linear,
+			pausable: true,
+			paused: false,
+		};
+		assert_noop!(
+			LBPPallet::validate_pool_data(&pool_data),
+			Error::<Test>::MaxSaleDurationExceeded
+		);
+	});
+}
+
+#[test]
+fn create_pool_should_work() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(LBPPallet::create_pool(
+			Origin::root(),
+			ALICE,
+			LBPAssetInfo {
+				id: ACA,
+				amount: 1_000_000_000,
+				initial_weight: 20,
+				final_weight: 90,
+			},
+			LBPAssetInfo {
+				id: DOT,
+				amount: 2_000_000_000,
+				initial_weight: 80,
+				final_weight: 10,
+			},
+			(10u64, 20u64),
+			WeightCurveType::Linear,
+			true,
+		));
+
+		assert_eq!(Currency::free_balance(ACA, &ACA_DOT_POOL_ID), 1_000_000_000);
+		assert_eq!(Currency::free_balance(DOT, &ACA_DOT_POOL_ID), 2_000_000_000);
+		assert_eq!(
+			Currency::free_balance(ACA, &ALICE),
+			INITIAL_BALANCE.saturating_sub(1_000_000_000)
+		);
+		assert_eq!(
+			Currency::free_balance(DOT, &ALICE),
+			INITIAL_BALANCE.saturating_sub(2_000_000_000)
+		);
+		assert_eq!(Currency::reserved_balance(HDX, &ALICE), POOL_DEPOSIT);
+		assert_eq!(
+			Currency::free_balance(HDX, &ALICE),
+			INITIAL_BALANCE.saturating_sub(POOL_DEPOSIT)
+		);
+		assert_eq!(LBPPallet::pool_deposit(&ACA_DOT_POOL_ID), POOL_DEPOSIT);
+
+		assert_eq!(LBPPallet::get_pool_assets(&ACA_DOT_POOL_ID).unwrap(), vec![ACA, DOT]);
+
+		// verify that `last_weight_update`, `last_weights` and `paused` fields are correctly initialized
+		let updated_pool_data = LBPPallet::pool_data(ACA_DOT_POOL_ID);
+		assert_eq!(updated_pool_data.last_weight_update, 0);
+		assert_eq!(updated_pool_data.last_weights, ((ACA, 20), (DOT, 80)));
+		assert_eq!(updated_pool_data.paused, true);
+
```

Comment:
it might be good check all the storage changes after pool created - so PoolOwner, Balacnes are missing here.